spec/unit/virtus/attribute/collection/value_coerced_predicate_spec.rb

```diff
@@ -0,0 +1,31 @@
+require 'spec_helper'
+require 'set'
+
+describe Virtus::Attribute::Collection, '#value_coerced?' do
+  subject { object.value_coerced?(input) }
+
+  let(:object) { described_class.build(Array[Integer]) }
+
+  context 'when input has correctly typed members' do
+    let(:input) { [1,2,3] }
```

Comment:
@Bartuz By making it part of the build, like here: https://github.com/dblock/ruby-enum or in larger projects, https://github.com/ruby-grape/grape